---
title: "OPFs v2"
author: "Kathryn Iverson"
date: "February 3rd, 2016"
output: ioslides_presentation
---
```{r echo=F, message=FALSE, warning=FALSE}
library(RSvgDevice)
library(ggplot2)
library(gplots)
library(reshape)
```

## OPFs:

### What:

- Operational protein families are clusters of similar proteins
- Creating OPFs is a database free method of grouping sequences 
- Analogous to OTUs

### Why:

Databases are not great

* Genes present are limited to what we "know" or at least have seen before
* Annotations are limited in the same way
* Leaves a large (often over half) of the data in a black box


## Quickly:

1. Assemble
1. ORF/gene prediction
1. Get counts
1. All v all BLAST
1. Cluster


## Estimate gene counts

Map reads with bowtie (and normalize)

### Counts file

* Use bowtie to map reads to genes
* Normalize by sequence length 

```
ceiling(num_reads_mapped * 100 / len(gene))
````

## Creating OPFs

### BLAST

All v all

Blast all genes against each other (proteins, blastp)

```
-max_target_seqs 10000
```

### Cluster

mg-cluster with mothur

```
mgcluster(blast=allVall100.out, count=allgenes.counts)
```

## Datasets 

1. Obesity in twins (Turnbaugh et al. 2009)
    - 18 metagenomes from lean and obese twin pairs and their mothers
    - From MG-RAST
1. Pregnancy (Koren et al. 2012)
    - 20 metagenomes from mothers during the first and third trimester of pregnancy
    - From NCBI (MG-RAST data is messed up and being fixed with zero urgency)

## Datasets {.smaller} 

HMP data

- Anterior nares
    - 87 metagenomes
- Buccal mucosa
    - 107 metagenomes
- Posterior fornix
    - 51 metagenomes
- Supragingival plaque
    - 118 metagenomes
- Stool
    - 138 metagenomes
- Tongue dorsum
    - 128 metagenomes

## Twin data

```{r echo=FALSE, message=FALSE, warning=FALSE}
summarySE <- function(data=NULL, measurevar, groupvars=NULL, na.rm=FALSE,
                      conf.interval=.95, .drop=TRUE) {
    require(plyr)

    # New version of length which can handle NA's: if na.rm==T, don't count them
    length2 <- function (x, na.rm=FALSE) {
        if (na.rm) sum(!is.na(x))
        else       length(x)
    }

    # This is does the summary; it's not easy to understand...
    datac <- ddply(data, groupvars, .drop=.drop,
                   .fun= function(xx, col, na.rm) {
                           c( N    = length2(xx[,col], na.rm=na.rm),
                              mean = mean   (xx[,col], na.rm=na.rm),
                              sd   = sd     (xx[,col], na.rm=na.rm)
                              )
                          },
                    measurevar,
                    na.rm
             )

    # Rename the "mean" column    
    datac <- rename(datac, c("mean"=measurevar))

    datac$se <- datac$sd / sqrt(datac$N)  # Calculate standard error of the mean

    # Confidence interval multiplier for standard error
    # Calculate t-statistic for confidence interval: 
    # e.g., if conf.interval is .95, use .975 (above/below), and use df=N-1
    ciMult <- qt(conf.interval/2 + .5, datac$N-1)
    datac$ci <- datac$se * ciMult

    return(datac)
}

t.dists <- read.csv(file='data/references/distance_comparison.csv', sep=',', header=T)
dfc <- summarySE(t.dists, measurevar="dist_bars", groupvars=c("type","method"))

ggplot(dfc, aes(x=method, y=dist_bars, group=type, fill=type)) + 
    geom_errorbar(aes(ymin=dist_bars-se, ymax=dist_bars+se), width=.1, position=position_dodge(.9)) +
    geom_bar(position=position_dodge(), stat='identity', colour='black') + theme_bw() + ylab('Distance') + xlab(' ') + scale_x_discrete(limits=c("unifrac", "unifrac_uw", "16s","opf", "kegg"), labels=c('Weighted Unifrac', 'Unweighted Unifrac', '16S', 'OPF', 'KEGG')) + theme(legend.justification=c(0,1), legend.position=c(0,1)) + scale_fill_manual(values=c("#000000", '#666666', "white"), 
                       name="Comparison",
                       breaks=c("all", "family", "twins"),
                       labels=c("Unrelated", "Inside family", "Twin to twin"))
```


## Pregnancy data

KEGG masks variation, notice scale

```{r echo=FALSE, message=FALSE, warning=FALSE}
preg.pcoa <- read.csv(file='data/references/pregnancy_pcoa_vars.csv', sep=',', header=T)
ggplot(preg.pcoa, aes(axis1, axis2, shape=trimester)) + geom_point(size=5) + theme_bw()  + facet_wrap(~method, ncol=1, scales='free')
```

<!--
## Shared twin OPFs, OTUs and KEGG categories

```{r echo=FALSE}

t.bin <- read.table(file='/Users/kiverson/labmeetingslides/opfs_v2/data/references/twin_opf_bin.csv', sep=' ', header=T, row.names=1)
rsum <- rowSums(t.bin)
t.bin['sum'] <- NA
t.bin$sum <- rsum
twin.o.sums <- matrix(nrow=18, ncol=1)
 
for (i in 1:18){
    s <- sum(t.bin$sum==i)
    twin.o.sums[i,1] <- s
}
tk.bin <- read.table(file='/Users/kiverson/labmeetingslides/opfs_v2/data/references/twin_kegg_bin.csv', sep=' ', header=T, row.names=1)
rksum <- rowSums(tk.bin)
tk.bin['sum'] <- NA
tk.bin$sum <- rksum
twin.k.sums <- matrix(nrow=18, ncol=1)
 
for (i in 1:18){
    s <- sum(tk.bin$sum==i)
 	twin.k.sums[i,1] <- s
}
ts.bin <- read.table(file='/Users/kiverson/labmeetingslides/opfs_v2/data/references/twin_16s_bin.csv', sep=' ', header=T, row.names=1)
rssum <- rowSums(ts.bin)
ts.bin['sum'] <- NA
ts.bin$sum <- rssum
twin.s.sums <- matrix(nrow=18, ncol=1)
 
for (i in 1:18){
 	s <- sum(ts.bin$sum==i)
 	twin.s.sums[i,1] <- s
}

plot(twin.o.sums, log='y', ylim=c(0.6, 6000), xlab='# of samples', ylab='Shared OPFs/OTUs/KEGG Categories', pch=15, cex=4)
points(twin.s.sums, pch=16, cex=4)
points(twin.k.sums, pch=17, cex=4)
legend("bottomleft", c('OPF', 'OTU', 'KEGG'), pch=c(15, 16, 17),)

```
-->

## HMP Anterior nares

```{r echo=FALSE}

#plot
par(mfrow=c(1,2))

anterior_nares.ko.axes<-read.table("data/references/anterior_nares.ko.merge.thetayc.1.lt.nmds.axes")
anterior_nares.opf.axes<-read.table("data/references/anterior_nares.opf.merge.thetayc.1.lt.nmds.axes")

plot(anterior_nares.ko.axes, main="KO")
plot(anterior_nares.opf.axes, main="OPFs")

```

## HMP Posterior fornix

```{r echo=FALSE}
par(mfrow=c(1,2))
posterior_fornix.ko.axes<-read.table("data/references/posterior_fornix.ko.merge.thetayc.1.lt.nmds.axes")
posterior_fornix.opf.axes<-read.table("data/references/posterior_fornix.opf.merge.thetayc.1.lt.nmds.axes")

plot(posterior_fornix.ko.axes, main="KO")
plot(posterior_fornix.opf.axes, main="OPFs")

```

## HMP Buccal mucosa

```{r echo=FALSE}
par(mfrow=c(1,2))
buccal_mucosa.ko.axes<-read.table("data/references/buccal_mucosa.ko.merge.thetayc.1.lt.nmds.axes")
buccal_mucosa.opf.axes<-read.table("data/references/buccal_mucosa.opf.merge.thetayc.1.lt.nmds.axes")

plot(buccal_mucosa.ko.axes, main="KO")
plot(buccal_mucosa.opf.axes, main="OPFs")
```

## Where do we go from here?

- OPFs are a unit
- As an OTU analog can do similar analysis
- Re-purpose diversity metrics
- OPF prediction and feature selection
